### ISSUE TITLE
moor_generator support build 2.0.0

### ISSUE THREAD

**Issue by vanlooverenkoen** [simolus3/moor#1100]
I am using some different generators. kiwi_generator, retrofit_generator, json_serializable. 

They are all using build 2.0.0 so for now I am stuck because the moor_generator is not yet supported to work with build 2.0.0

```
[WARNING] stderr: ../../../../.pub-cache/hosted/pub.dartlang.org/moor_generator-4.1.0/lib/src/backends/build/preprocess_builder.dart:74:51: Error: The argument type 'String' can't be assigned to the parameter type 'Uri'.
[WARNING] stderr:  - 'Uri' is from 'dart:core'.
[WARNING] stderr:               .map((stmt) => AssetId.resolve(stmt.importedFile, from: asset))
[WARNING] stderr:                                                   ^
[WARNING] stderr: ../../../../.pub-cache/hosted/pub.dartlang.org/moor_generator-4.1.0/lib/src/backends/build/build_backend.dart:23:39: Error: The argument type 'String' can't be assigned to the parameter type 'Uri'.
[WARNING] stderr:  - 'Uri' is from 'dart:core'.
[WARNING] stderr:     final from = AssetId.resolve(base.toString());
[WARNING] stderr:                                       ^
[WARNING] stderr: ../../../../.pub-cache/hosted/pub.dartlang.org/moor_generator-4.1.0/lib/src/backends/build/build_backend.dart:24:28: Error: The argument type 'String' can't be assigned to the parameter type 'Uri'.
[WARNING] stderr:  - 'Uri' is from 'dart:core'.
[WARNING] stderr:     return AssetId.resolve(import, from: from).uri;
[WARNING] stderr:                            ^
[WARNING] stderr: ../../../../.pub-cache/hosted/pub.dartlang.org/moor_generator-4.1.0/lib/src/backends/build/build_backend.dart:40:32: Error: The argument type 'String' can't be assigned to the parameter type 'Uri'.
[WARNING] stderr:  - 'Uri' is from 'dart:core'.
[WARNING] stderr:     return AssetId.resolve(uri.toString(), from: step.inputId);
[WARNING] stderr:                                ^
[WARNING] stderr: ../../../../.pub-cache/hosted/pub.dartlang.org/moor_generator-4.1.0/lib/src/backends/build/serialized_types.dart:134:67: Error: The argument type 'String' can't be assigned to the parameter type 'Uri'.
[WARNING] stderr:  - 'Uri' is from 'dart:core'.
[WARNING] stderr:           await buildStep.resolver.libraryFor(AssetId.resolve(uri.toString()));
[WARNING] stderr:     
```

**Comment by vanlooverenkoen** [comment]
I got all the errors fixed except this one

![image](https://user-images.githubusercontent.com/21172855/111317896-d8f94b00-8664-11eb-886f-17c462bde967.png)

**Comment by simolus3** [comment]
I've already done some work to support the latest `build` version on my `develop` branch. I'll try to release that this week.

**Comment by vanlooverenkoen** [comment]
Ah damn I just added support for the latest build version

**Comment by vanlooverenkoen** [comment]
I will use my branch for now. No problem

**Comment by UliPrantz** [comment]
Also besides that it would be really helpful to also upgrade the other dependencies which interact with `build` like `json_annotations`. That actually shouldn't be a problem since `json_annotations` >= 4.0.0 already depend on the new `build` 2.0.0. So once `build` is updated it would be really nice if you could update `json_annotations` too.

Anyways thanks for your great work and this cool package :)

**Comment by vanlooverenkoen** [comment]
Yeah that is required as well indeed

**Comment by UliPrantz** [comment]
Wow thanks for that really quick response.

**Comment by vanlooverenkoen** [comment]
The power of mobile application with push notifications 😀😀

**Comment by anisalibegic** [comment]
> I've already done some work to support the latest `build` version on my `develop` branch. I'll try to release that this week.

Can I use that branch until it's release on master and pub.dev?

**Comment by RakaAlrian** [comment]
I tried this in my pubspec.yaml 
```
moor_generator:
    git:
      url: https://github.com/simolus3/moor.git
      ref: develop
      path: moor/moor_generator
```
but then I got this error
>pub get failed (1; Could not find a file named "moor/moor_generator/pubspec.yaml" in https://github.com/simolus3/moor.git 1d08b73ed442910774638440f9d9d0277559179b.)

any suggestion please?

**Comment by anisalibegic** [comment]
`moor_generator:
    git:
      url: https://github.com/simolus3/moor.git
      ref: develop
      path: moor_generator`

**Comment by simolus3** [comment]
I've just published `moor_generator` version 4.2.0 (nice!), which supports the latest `build`, `source_gen`, `analyzer` and `dart_style` packages.
